pandas/core/indexes/multi.py

```diff
@@ -50,6 +50,43 @@
          target_klass='MultiIndex or list of tuples'))
 
 
+class MultiIndexUIntEngine(libindex.BaseMultiIndexCodesEngine,
+                           libindex.UInt64Engine):
+    """
+    Manage a MultiIndex by mapping label combinations to positive integers.
+    """
+    _base = libindex.UInt64Engine
+
+    def codes_to_ints(self, codes):
```

Comment:
add to the base class a a NotImplementedError

Reply:
_which_ base class? Anyway, I don't see the point, as ``NotImplementedError`` means you might want to implement it, which is not the case here. I'd rather make ``BaseMultiIndexCodesEngine`` not instantiable if this is your worry.